File: c/b2501.c
#include <stdio.h>

// 2501: 약수 구하기

int main(void) {
	int N, K, i;
	int count = 0;
	int result = 0;

	scanf("%d %d", &N, &K);

	for (i = 1; i <= N; i++) {
		if (N % i == 0) {
			count++;
			if (count == K) {
				result = i;
				break;
			}
		}
	}

	printf("%d\n", result);
	return 0;
}

File: python/b2501.py

# 2501: 약수 구하기

inputs = input().split(" ")
N = int(inputs[0])
K = int(inputs[1])

count = 0
result = 0
for i in range(1, N+1):
	if N % i == 0:
		count += 1
		if count == K:
			result = i
			break

print(result)